src/Surfnet/StepupRa/RaBundle/Security/Authentication/Token/SamlToken.php

```diff
@@ -39,18 +39,19 @@ class SamlToken extends AbstractToken
      */
     private $institutionConfigurationOptions;
 
-    public function __construct(
-        Loa $loa,
-        array $roles = [],
-        InstitutionConfigurationOptions $institutionConfigurationOptions = null
-    ) {
+    public function __construct(Loa $loa, array $roles = [])
```

Comment:
imho this is the wrong approach. The "scoping" of a token by `InstitutionConfigurationOptions` still is a requirement, a token cannot possibly be valid without the `InstitutionConfigurationOptions`. 

Think about it this way: the 99.9% usecase is that the `InstitutionConfigurationOptions` is set once, never modified. In a single exception, and Identity with SRAA role, the `InstitutionConfigurationOptions` may be overwritten, like the institution of the identity. Specific functionality for a specifc usecase. Now it is build as a generic solution for all, because we had to modify the `InstitutionConfigurationOptions` for a specific usecase. This line of reasoning can be found in the difference of the setter as well: there is a guard missing that this override may only happen if the identity has the SRAA role.

Alternative suggestion: create a `changeInstitutionScope(Institution $scopedInstitution, InstitutionConfigurationOptions $scopedOptions)` method on the token, add the guard that it may only be invoked if the Identity has the SRAA role (exception if not), and only change it there. Explicit, guarded behavior, testable, and has the added benefit of cleaning up controller code as well :smile: (aside: commandHandler would be nice)